tests/providers/aws/services/eks/eks_cluster_uses_a_supported_version/eks_cluster_uses_a_supported_version.py

```diff
@@ -0,0 +1,174 @@
+from unittest import mock
+
+from prowler.providers.aws.services.eks.eks_service import EKSCluster
+from tests.providers.aws.utils import AWS_ACCOUNT_NUMBER, AWS_REGION_EU_WEST_1
+
+cluster_name = "cluster_test"
+cluster_arn = (
+    f"arn:aws:eks:{AWS_REGION_EU_WEST_1}:{AWS_ACCOUNT_NUMBER}:cluster/{cluster_name}"
+)
+
+
+class Test_eks_cluster_ensure_version_is_supported:
+    def test_no_clusters(self):
+        eks_client = mock.MagicMock
+        eks_client.clusters = []
+        with mock.patch(
+            "prowler.providers.aws.services.eks.eks_service.EKS",
+            eks_client,
+        ):
+            from prowler.providers.aws.services.eks.eks_cluster_uses_a_supported_version.eks_cluster_uses_a_supported_version import (
+                eks_cluster_uses_a_supported_version,
+            )
+
+            check = eks_cluster_uses_a_supported_version()
+            result = check.execute()
+            assert len(result) == 0
+
+    def test_cluster_not_supported_verison(self):
+        eks_client = mock.MagicMock
+        eks_client.audit_config = {
+            "eks_cluster_supported_versions": ["1.28", "1.29", "1.30"]
+        }
+        eks_client.clusters = []
+        eks_client.clusters.append(
+            EKSCluster(
+                name=cluster_name,
+                version="1.22",
+                arn=cluster_arn,
+                region=AWS_REGION_EU_WEST_1,
+                logging=None,
+            )
+        )
+
+        with mock.patch(
+            "prowler.providers.aws.services.eks.eks_service.EKS",
+            eks_client,
+        ):
+            from prowler.providers.aws.services.eks.eks_cluster_uses_a_supported_version.eks_cluster_uses_a_supported_version import (
+                eks_cluster_uses_a_supported_version,
+            )
+
+            check = eks_cluster_uses_a_supported_version()
+            result = check.execute()
+            assert len(result) == 1
+            assert result[0].status == "FAIL"
+            assert (
+                result[0].status_extended
+                == f"EKS cluster {cluster_name} must have a version of ['1.28', '1.29', '1.30']"
+            )
+            assert result[0].resource_id == cluster_name
+            assert result[0].resource_arn == cluster_arn
+            assert result[0].resource_tags == []
+            assert result[0].region == AWS_REGION_EU_WEST_1
+
+    def test_cluster_supported_verison1(self):
```

Comment:
Please do not use test names like 1, 2, 3... use more descriptive names instead. For example, `test_eks_cluster_using_version_1_XX`